src/hardware/serialport/softmodem.cpp

```diff
@@ -234,10 +250,8 @@ void CSerialModem::Reset(){
 	oldDTRstate = getDTR();
 	flowcontrol = 0;
 	plusinc = 0;
-	if(clientsocket) {
-		delete clientsocket;
-		clientsocket=0;
-	}
+	clientsocket.reset(nullptr);
```

Comment:
Nice :) The `unique_ptr`s made this code easier to follow.

Reply:
Thanks! I'm starting to see a path toward refurbishing most of the code.  I will probably double-back on my audio code for another round  one of these days, after a bit more work on the CI scripts.